R/BoostedStacking.R

```diff
@@ -0,0 +1,352 @@
+        new.feat = new.feat, feat.name = paste0("feat.", i), td)
+    }
+  }
+  if (sm.pt == "prob") {
+    if (bms.pt == "prob") {
+      return(as.matrix(getPredictionProbabilities(newest.pred, cl = td$class.levels)))
+    } else { # if bms.pt="response" and sm.pt="prob" predict must be repeated
+      last.model = .model$learner.model$base.models[[niter]]
+      last.model$learner$predict.type = "prob"
+      newest.pred = predict(last.model, newdata = new.data[, -ncol(new.data)]) #FIXMENOW
+      return(as.matrix(getPredictionProbabilities(newest.pred, cl = td$class.levels)))
+    }
+  } else { #sm.pt = "response" and  bms.pt = "prob"/"response"
+    return(getPredictionResponse(newest.pred)) #
+  }
+}
+
+
+#' Creates x best learners from ModelMultiplexer TuneResult.
+#' 
+#' @param tune.result [\code{\link{TuneResult}}]\cr
+#'   \code{TuneResult} from \code{ModelMultiplexer}
+#' @param model.multiplexer [\code{\link{ModelMultiplexer}}]\cr
+#'   Needed to exract fix parameters.
+#' @param mm.ps [\code{list containing \link{ParamSet}}]\cr
+#'   Needed to extract fix parameters.
+#' @param x.best [\code{integer(1)}]
+#'   Number of best learners to extract.
+#' @param measure [\code{\link{Measure}}]\cr
+#'   One Measure to apply to "extract from best".
+#' @return [\code{list of x best learners}].
+#' @examples 
+#' \dontrun{
+#' tsk = pid.task
+#' lrns = list(
+#'   makeLearner("classif.gbm", distribution = "bernoulli"),
+#'   makeLearner("classif.randomForest", ntree = 3))
+#' bpt = spt = "prob"
+#' lrns = lapply(lrns, setPredictType, bpt)
+#' mm = makeModelMultiplexer(lrns)
+#' ctrl = makeTuneControlRandom(maxit = 3L)
+#' ps = makeModelMultiplexerParamSet(mm,
+#'   makeIntegerParam("n.trees", lower = 1L, upper = 3L),
+#'   makeIntegerParam("interaction.depth", lower = 1L, upper = 2L),
+#'   makeIntegerParam("ntree", lower = 1L, upper = 3L),
+#'   makeIntegerParam("mtry", lower = 1L, upper = getTaskNFeats(tsk)))
+#' tune.res = tuneParams(learner = mm, task = tsk, resampling = cv2, par.set = ps, control = ctrl)
+#' best = makeXBestLearnersFromMMTuneResult(tune.result = tune.res, 
+#'   model.multiplexer = mm, mm.ps = ps, x.best = 2, measure  =mmce)
+#' best
+#' }
+
+makeXBestLearnersFromMMTuneResult = function(tune.result, model.multiplexer, mm.ps, x.best = 5, measure) {
+  # checks
+  assertClass(tune.result, "TuneResult")
+  assertClass(model.multiplexer, "ModelMultiplexer")
+  assertClass(measure, "Measure")
+  assertInt(x.best, lower = 1)
+  # body
+  measure.name = paste0(measure$id, ".test.mean")
+  opt.grid = as.data.frame(trafoOptPath(tune.result$opt.path), stringsAsFactors = FALSE)
+  #FIXME minimize
+  if (measure$minimize) {
+    ord = order(opt.grid[, measure.name])[1:x.best]
+  } else {
+    ord = rev(order(opt.grid[, measure.name]))[1:x.best]
+  }
+  opt.grid = opt.grid[ord, ]
+  j = sapply(opt.grid, is.factor)
+  opt.grid[j] = lapply(opt.grid[j], as.character)
+  # checks2
+  if(NROW(opt.grid) < x.best) {
```

Comment:
I see sometimes `if(...)` and `if (...)`. Just use always whitespace between if and braces, e.g. `if (...)`.